graphcanvas/tests/test_graph_node_hover_tool.py

```diff
@@ -0,0 +1,88 @@
+from StringIO import StringIO
+import sys
+import unittest
+
+import networkx
+
+from enable.api import BasicEvent
+from graphcanvas.graph_node_hover_tool import GraphNodeHoverTool
+from graphcanvas.graph_container import GraphContainer
+from graphcanvas.graph_node_component import GraphNodeComponent
+
+
+def simple_callback(label):
+    print label
+
+
+class TestGraphNodeHoverTool(unittest.TestCase):
+    def setUp(self):
+        g = networkx.DiGraph()
+        self.container = GraphContainer(graph=g)
+        self.tool = GraphNodeHoverTool(component=self.container,
+                                       callback=simple_callback)
+        self.container.tools.append(self.tool)
+        self.container.components.append(
+            GraphNodeComponent(position=[0, 0],
+                               value='test')
+        )
+
+    def tearDown(self):
+        del self.container
+        del self.tool
+
+    def test__is_in(self):
+        self.assertTrue(self.tool._is_in(0, 0))
+        self.assertFalse(self.tool._is_in(-100, -100))
+
+    def test_normal_mosuse_move(self):
+        event = BasicEvent(x=10, y=10, handled=False)
+        self.tool.normal_mouse_move(event)
+        self.assertEqual(self.tool._last_xy, (10, 10))
+
+    def test_on_hover(self):
+        # test in
+        self.tool._last_xy = (0, 0)
+        # capture stdout
+        stdout = sys.stdout
```

Comment:
hmm..., i see now why you use the stdout. I would replace the `callback` at the tool with a mock.Mock(), (the mock module). You can query the mock object if it has been called and how many times